crates/transaction-pool/src/lib.rs

```diff
@@ -350,7 +350,8 @@ where
         transaction: Self::Transaction,
     ) -> PoolResult<TxHash> {
         let (_, tx) = self.validate(origin, transaction).await;
-        self.pool.add_transactions(origin, std::iter::once(tx)).pop().expect("exists; qed")
+        let mut results = self.pool.add_transactions(origin, std::iter::once(tx));
+        results.pop().expect("result length is the same as the input")
```

Comment:
not sure what this error message means, where are we guaranteed that `transactions` ins't zero? anyhow out of scope of this pr probably

Reply:
add_transaction takes in an iterator, the output is a Vec<Result<_>> collected from the iterator so it has the same length